data/transaction/construction_test.go

```diff
@@ -0,0 +1,145 @@
+		SndAddr:  getPubkeyOfAddress(t, "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th"),
+		GasPrice: 1000000000,
+		GasLimit: 100000,
+		Data:     []byte("for the book"),
+		ChainID:  []byte("local-testnet"),
+		Version:  1,
+	}
+
+	tx.Signature = signTransaction(t, alicePrivateKeyHex, tx)
+	require.Equal(t, "9074789e0b4f9b2ac24b1fd351a4dd840afcfeb427b0f93e2a2d429c28c65ee9f4c288ca4dbde79de0e5bcf8c1a5d26e1b1c86203faea923e0edefb0b5099b0c", hex.EncodeToString((tx.Signature)))
+
+	data, _ := contentMarshalizer.Marshal(tx)
+	require.Equal(t, "085b1209008ac7230489e800001a208049d639e5a6980d1cd2392abcce41029cda74a1563523a202f09641cc2618f82a200139472eff6886771a982f3083da5d421f24c29181e63888228dc81ca60d69e1388094ebdc0340a08d064a0c666f722074686520626f6f6b520d6c6f63616c2d746573746e6574580162409074789e0b4f9b2ac24b1fd351a4dd840afcfeb427b0f93e2a2d429c28c65ee9f4c288ca4dbde79de0e5bcf8c1a5d26e1b1c86203faea923e0edefb0b5099b0c", hex.EncodeToString((data)))
+
+	txHash := contentHasher.Compute(string(data))
+	require.Equal(t, "af53e0fc86612d5068862716b5169effdf554951ecc89849b0e836eb0b63fa3e", hex.EncodeToString((txHash)))
+}
+
+func TestConstructTransaction_WithDataWithLargeValue(t *testing.T) {
+	tx := &Transaction{
+		Nonce:    92,
+		Value:    stringToBigInt("123456789000000000000000000000"),
+		RcvAddr:  getPubkeyOfAddress(t, "erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx"),
+		SndAddr:  getPubkeyOfAddress(t, "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th"),
+		GasPrice: 1000000000,
+		GasLimit: 100000,
+		Data:     []byte("for the spaceship"),
+		ChainID:  []byte("local-testnet"),
+		Version:  1,
+	}
+
+	tx.Signature = signTransaction(t, alicePrivateKeyHex, tx)
+	require.Equal(t, "39938d15812708475dfc8125b5d41dbcea0b2e3e7aabbbfceb6ce4f070de3033676a218b73facd88b1432d7d4accab89c6130b3abe5cc7bbbb5146e61d355b03", hex.EncodeToString((tx.Signature)))
+
+	data, _ := contentMarshalizer.Marshal(tx)
+	require.Equal(t, "085c120e00018ee90ff6181f3761632000001a208049d639e5a6980d1cd2392abcce41029cda74a1563523a202f09641cc2618f82a200139472eff6886771a982f3083da5d421f24c29181e63888228dc81ca60d69e1388094ebdc0340a08d064a11666f722074686520737061636573686970520d6c6f63616c2d746573746e65745801624039938d15812708475dfc8125b5d41dbcea0b2e3e7aabbbfceb6ce4f070de3033676a218b73facd88b1432d7d4accab89c6130b3abe5cc7bbbb5146e61d355b03", hex.EncodeToString((data)))
+
+	txHash := contentHasher.Compute(string(data))
+	require.Equal(t, "e4a6048d92409cfe50f12e81218cb92f39966c618979a693b8d16320a06061c1", hex.EncodeToString((txHash)))
+}
+
+func stringToBigInt(input string) *big.Int {
+	result := big.NewInt(0)
+	_, _ = result.SetString(input, 10)
+	return result
+}
+
+func getPubkeyOfAddress(t *testing.T, address string) []byte {
+	pubkey, err := addressEncoder.Decode(address)
+	require.Nil(t, err)
+	return pubkey
+}
+
+func signTransaction(t *testing.T, senderSeedHex string, tx *Transaction) []byte {
```

Comment:
I would rename it to `processTransactionSignature` or so because, with the current naming, I would think that the transaction is being signed within the function (and maybe the transaction is returned)

Reply:
Renamed to `computeTransactionSignature`.